android/app/src/main/java/org/fossasia/openevent/viewmodels/auth/ChangePasswordActivityViewModel.java

```diff
@@ -0,0 +1,101 @@
+package org.fossasia.openevent.viewmodels.auth;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.auth.Login;
+import org.fossasia.openevent.data.auth.User;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.utils.AuthUtil;
+import org.fossasia.openevent.utils.JWTUtils;
+import org.json.JSONException;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class ChangePasswordActivityViewModel extends ViewModel {
+
+    public static final int ON_ERROR = 2;
+    public static final int ON_COMPLETE = 3;
+
+    private RealmDataRepository realmRepo;
+    private User user;
+    private LiveData<User> userLiveData;
+    private MutableLiveData<Integer> checkPasswordResponse;
+    private MutableLiveData<Integer> changePasswordResponse;
+    private final CompositeDisposable compositeDisposable;
+
+    public ChangePasswordActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+    }
+
+    public LiveData<User> getUser() {
+        if (userLiveData == null) {
+            user = realmRepo.getUser();
+            user.addChangeListener(realmModel -> {});
+            userLiveData = RealmDataRepository.asLiveDataForObject(user);
+        }
+        return userLiveData;
+    }
+
+    public LiveData<Integer> checkCurrentPassword(String currentPassword) {
+        if (checkPasswordResponse == null) {
+            checkPasswordResponse = new MutableLiveData<>();
+        }
+        
+        compositeDisposable.add(APIClient.getOpenEventAPI().login(new Login(user.getEmail(), currentPassword))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(loginResponse -> {
+                            Timber.d("Received token. Password is correct");
+                        },
+                        throwable -> {
+                            checkPasswordResponse.setValue(ON_ERROR);
+                            Timber.d("Error changing password: " + throwable.getMessage());
+                        },
+                        () -> {
+                            checkPasswordResponse.setValue(ON_COMPLETE);
+                        }));
+        return checkPasswordResponse;
+    }
+
+    public LiveData<Integer> changePassword(String newPassword) {
+        if (changePasswordResponse == null) {
+            changePasswordResponse = new MutableLiveData<>();
+        }
+        int id = 0;
+        try {
+            id = JWTUtils.getIdentity(AuthUtil.getAuthorization());
+        } catch (JSONException e) {
+            e.printStackTrace();
```

Comment:
You should not handle error this way. Send the error immediately to View and log it using Timber. Because if there's an error here, id would be wrong already. No point in continuing, so move the entire code in try or return early